server/src/main/java/de/uftos/dto/responsedtos/StudentGroupResponseDto.java

```diff
@@ -5,22 +5,41 @@
 import de.uftos.entities.StudentGroup;
 import de.uftos.entities.Subject;
 import de.uftos.entities.Tag;
+import jakarta.validation.constraints.NotNull;
+
 import java.util.Collections;
 import java.util.List;
 
-public record StudentGroupResponseDto(String id,
-                                      String name,
-                                      List<Student> students,
-                                      List<GradeResponseDto> grades,
-                                      List<Tag> tags,
-                                      List<Lesson> lessons,
-                                      List<Subject> subjects) {
+/**
+ * A data transfer object used in the student group HTTP requests.
+ *
+ * @param id       the id of the student group
+ * @param name     the name of the student group
+ * @param students the students in the student group
+ * @param grades   the grades of the student group
+ * @param tags     the tags of the student group
+ * @param lessons  the lessons of the student group
+ * @param subjects the subjects of the student group
+ */
+public record StudentGroupResponseDto(@NotNull String id,
+                                      @NotNull String name,
```

Comment:
Those should probably be `@NotEmpty`

Reply:
`NotNull` is used in every response dto.
If you change it here you could argue, that all others need to be changed as well.

Reply:
Nevermind im just blind